lib/session.js

```diff
@@ -9,8 +9,16 @@ TinyCertManagement = require('./cert.js');
 var TinyCertSession = function TinyCertSession(apiKey) {
     var sessionToken = null;
 
+    var encode = function encode(val) {
+        var transform = function transform(ch) {
+            return '%' + ch.charCodeAt(0).toString(16).toUpperCase();
+        }
+
+        return encodeURIComponent(val).toString().replace(/([^a-zA-Z0-9-_.%])/g, transform).replace('%20', '+');
```

Comment:
This will stop substitution at the first instance of %20 - so `a%20b%20c` will become `a+b%20c` which is incorrect.   It should be `a+b+c`.   This is easily fixed by calling `.replace` with a RegExp:
```js
      return encodeURIComponent(val).toString().replace(/([^a-zA-Z0-9-_.%])/g, transform)
        .replace(new RegExp('%20', 'g'), '+')
```